server/src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -224,9 +226,27 @@ protected Configuration readConfiguration(ServletContext context)
         // load the defaults
         MapConfiguration defaults = new MapConfiguration(ConfigProperties.DEFAULT_PROPERTIES);
 
-        // merge the defaults with the system configuration. ORDER MATTERS.
-        // system config must be read FIRST otherwise settings won't be applied.
-        return EncryptedConfiguration.merge(systemConfig, defaults);
+        // Default to Postgresql if jpa.config.hibernate.dialect is unset
+        DatabaseConfigFactory.SupportedDatabase db = determinDatabaseConfiguration(systemConfig.getString
+            ("jpa.config.hibernate.dialect", PostgreSQLDialect.class.getName()));
+        log.info("Running under {}", db.getLabel());
+        Configuration databaseConfig = DatabaseConfigFactory.fetchConfig(db);
+
+        // merge the defaults with the system configuration. PARAMETER ORDER MATTERS.
+        // systemConfig must be FIRST to override subsequent configs.  This set up allows
+        // the systemConfig to override databaseConfig if necessary, but that's not really something
+        // users should be doing unbidden so it is undocumented.
+        return EncryptedConfiguration.merge(systemConfig, databaseConfig, defaults);
+    }
+
+    private DatabaseConfigFactory.SupportedDatabase determinDatabaseConfiguration(String dialect) {
+        if (dialect.matches("(?i)" + DatabaseConfigFactory.SupportedDatabase.MYSQL.getLabel())) {
```

Comment:
What is the advantage of using a regex here instead of .equalsIgnoreCase? Also, I was under the impression that when you do inline case-insensitivity, it only applies to the group in which the modifier appears. I could be wrong, but this seems like it's just doing a normal comparison.